pkgs/applications/graphics/pinta/default.nix

```diff
@@ -37,35 +40,39 @@ buildDotnetModule rec {
     owner = "PintaProject";
     repo = "Pinta";
     rev = version;
-    sha256 = "sha256-iOKJPB2bI/GjeDxzG7r6ew7SGIzgrJTcRXhEYzOpC9k=";
+    sha256 = "sha256-Bvzs1beq7I1+10w9pmMePqGCz2TPDp5UK5Wa9hbKERU=";
   };
 
-  postFixup = ''
-    # Rename the binary
-    mv $out/bin/Pinta $out/bin/pinta
+  # https://github.com/NixOS/nixpkgs/issues/38991
+  # bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
+  LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive";
 
-    # Copy desktop icons
-    for size in 16x16 22x22 24x24 32x32 96x96 scalable; do
-      mkdir -p $out/share/icons/hicolor/$size/apps
-      cp xdg/$size/* $out/share/icons/hicolor/$size/apps/
-    done
+  # Do the autoreconf/Makefile job manually
+  # TODO: use upstream build system
+  postBuild = ''
+    # Substitute translation placeholders
+    intltool-merge -x po/ xdg/pinta.appdata.xml.in xdg/pinta.appdata.xml
+    intltool-merge -d po/ xdg/pinta.desktop.in xdg/pinta.desktop
 
-    # Copy runtime icons
-    cp -r Pinta.Resources/icons/hicolor/16x16/* $out/share/icons/hicolor/16x16/
+    # Build translations
+    dotnet msbuild Pinta \
+      -p:ContinuousIntegrationBuild=true \
+      -p:Deterministic=true \
+      -target:CompileTranslations,PublishTranslations \
+      -p:BuildTranslations=true \
+      -p:PublishDir="$NIX_BUILD_TOP/source/publish"
```

Comment:
I tried using here instead:

- `dotnetBuildFlags = [ "-p:BuildTranslations=true" "-p:PublishDir=/build/source/publish" ]`
- `dotnetInstallFlags = [ "-p:BuildTranslations=true" "-p:PublishDir=/build/source/publish" ]`
- `dotnetFlags = [ "-p:BuildTranslations=true" "-p:PublishDir=/build/source/publish" ]`

However, none of them worked. As far I understand looking at the Pinta source code:

- `dotnetBuildFlags` will not work since this is not a `dotnet build` step
- `dotnetInstallFlags` will trigger a `dotnet publish` step and could work since this is what upstream [uses](https://github.com/PintaProject/Pinta/blob/2fcd1e0be544fb2ced5fb288f2313f696d83a5f2/.github/workflows/build.yml#L106), however we have a [single output](https://github.com/thiagokokada/nixpkgs/tree/4fd577c836ca39a2eb72e96d82d520ac8de47730/pkgs/build-support/build-dotnet-module/default.nix) during `installPhase` so it is not triggering the `CompileTranslations,PublishTranslations` target (or something else, I am not a specialist in dotnet)
- `dotnetFlags` is a last attempt since it adds the flags everywhere, however even them it didn't work.

CC @IvarWithoutBones.

Reply:
> dotnetInstallFlags will trigger a dotnet publish step and could work since this is what upstream uses

This is actually a bit different for us, since we seperate the build process with phases. `dotnet publish` is not allowed to compile anything, it can only install stuff. I think that's why it doesn't work here, since the translations need to be compiled.

> BTW, I didn't found any step in buildDotnetModule that triggers a dotnet msbuild, and I am also not sure the different between dotnet build and dotnet msbuild steps

`dotnet build` (and many other `dotnet` subcommands) trigger msbuild internally, while still allowing you to define msbuild properties. 

Looking at upstream, these targets should all run automatically. The following snippet might work? 
```nix
dotnetBuildFlags = [ "-p:BuildTranslations=true" ];
dotnetInstallFlags = [ "-p:BuildTranslations=true" "-p:PublishDir=/build/source/publish" ]; # Install target depends on `BuildTranslations` too
```

The `CompileTranslations` target runs [prior to `dotnet build`](https://github.com/PintaProject/Pinta/blob/8b1bcd8802ecdfae1cb85683d1574c26c23b6d6e/Pinta/Pinta.csproj#L35), so I think running it with `dotnet build` should be fine. 

It then gets installed [after `dotnet publish`](https://github.com/PintaProject/Pinta/blob/8b1bcd8802ecdfae1cb85683d1574c26c23b6d6e/Pinta/Pinta.csproj#L42), so I think we can set all flags it needs there as well. I'm not quite sure how the `PublishDir` flags works though, shouldn't that be set to `$out` in some way? It looks like files are directly copied there.

Reply:
I already tried setting as `dotnetBuildFlags`, however this also doesn't work. Neither setting both `dotnetBuildFlags` or `dotnetInstallFlags`, or even `dotnetFlags` that should set for every build step.

Reply:
> I'm not quite sure how the PublishDir flags works though, shouldn't that be set to $out in some way? It looks like files are directly copied there.

`PublishDir` seeems to be kinda of a temporary directory that stores some build artefacts. Them it copies the contents of `PublishDir` to the output.

Reply:
Really strange. Investigating the issue further it seems that the `CompileTranslations` and `PublishTranslations` targets are not building for some reason by default.

Reply:
Anyway, using `dotnet build` also worked here. Switching to it since `dotnet msbuild` is not documented it seems?